test/functional/README.md

```diff
@@ -0,0 +1,372 @@
+## Functional test suite
+
+This directory contains the functional test suite for the IoTA JSON. This test suite is based on mocha and chai. For
+mocks, we use the nock library. Additionally, it uses some specific functions to ease to implement the test. Helper
+functions are located in the `testUtils.js` file.
+
+There are 2 tests files in this directory:
+
+-   `fuctional-tests.js`: This file contains the test defined in the "classic way". This means, coded in the JS file as
+    any other mocha test. It uses the functions defined in the `testUtils.js` file to simplify the tests.
+-   `functional-tests-auto.js`: This file contains the test defined in the "automatic way". This means, the test cases
+    are defined as JSON in a separate file (`testCases.js`). This file is loaded by the test suite and the test cases
+    are automatically generated. This is the recommended way to define the test cases.
+
+### Automatic test cases
+
+Each test case is defined as a JSON object in the `testCases.js` file. This file is loaded by the test suite and the
+test cases are automatically generated. Each test case is defined as an object with the following elements:
+
+-   `describeName`: The name of the `DESCRIBE` test case. This will be used to generate the test case name in the mocha
+    test suite.
+-   `provision`: The JSON object that will be sent to the IoTA JSON provisioning API. This will be used to create the
+    group. It contains the following elements:
+    -   `url`: The URL of the provisioning API (group)
+    -   `method`: The HTTP method to use (POST)
+    -   `json`: The JSON object that defines the group
+    -   `headers`: The headers to send to the provisioning API. This should contain the `fiware-service` and
+        `fiware-servicepath` headers.
+    -   `skip`: optional. If set to `true`, the test case (`describe`) will be skipped. This is useful to skip test
+        cases that are not supported by the agent. It can also have a string value `lib`. This will skip the test case
```

Comment:
```suggestion
        cases that are not supported by the agent. It can also have a string value `"lib"`. This will skip the test case
```

Given that another possibility is boolean `true`, this way we make more explicit that this case referst to a string, not a boolean.